File: ios/BingMapsViewManager.m
#import <Foundation/Foundation.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(BingMapsViewManager, RCTViewManager)
    RCT_EXTERN_METHOD(setMapLocationFromManager:(nonnull NSNumber *)node location:(nonnull NSDictionary *)location);
    RCT_EXTERN_METHOD(setMapPinsFromManager:(nonnull NSNumber *)node pinData:(nonnull NSArray *)pinData);
    RCT_EXPORT_VIEW_PROPERTY(onMapPinClicked, RCTDirectEventBlock);
    RCT_EXPORT_VIEW_PROPERTY(onMapLoadingStatusChanged, RCTDirectEventBlock);
    RCT_EXPORT_VIEW_PROPERTY(pins, NSArray);
    RCT_EXPORT_VIEW_PROPERTY(mapLocation, NSDictionary);
    RCT_EXPORT_VIEW_PROPERTY(mapStyle, NSString);
    RCT_EXPORT_VIEW_PROPERTY(credentialsKey, NSString);
    RCT_EXPORT_VIEW_PROPERTY(buildingsVisible, BOOL);
    RCT_EXPORT_VIEW_PROPERTY(businessLandmarksVisible, BOOL);
    RCT_EXPORT_VIEW_PROPERTY(transitFeaturesVisible, BOOL);
    RCT_EXPORT_VIEW_PROPERTY(compassButtonVisible, BOOL);
    RCT_EXPORT_VIEW_PROPERTY(tiltButtonVisible, BOOL);
    RCT_EXPORT_VIEW_PROPERTY(zoomButtonsVisible, BOOL);
    RCT_EXPORT_VIEW_PROPERTY(copyrightDisplay, NSString);
@end

File: ios/BingMapsViewManager.swift
//
//  BingMapsViewManager.swift
//  BingMaps
//
//  Created by Akarsh Srivastava on 19/02/21.
//  Copyright © 2021 Facebook. All rights reserved.
//

import Foundation
import MicrosoftMaps
import MapKit

@objc(BingMapsViewManager)
class BingMapsViewManager: RCTViewManager {
  
  override static func requiresMainQueueSetup() -> Bool {
    return true;
  }
  
  override func view() -> UIView! {
    return BingMaps(frame: self.accessibilityFrame);
  }
  
  @objc func setMapPinsFromManager(_ node: NSNumber, pinData: NSArray) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
      ) as! BingMaps;
      component.setMapPins(pinData: pinData);
    }
  }
  
  
  @objc func setMapLocationFromManager(_ node: NSNumber, location: NSDictionary) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
        ) as! BingMaps;
      component.setMapLocation(location: location);
    }
  }
  
}
